pkg/repositories/models/execution.go

```diff
@@ -56,4 +56,10 @@ type Execution struct {
 	// The user responsible for launching this execution.
 	// This is also stored in the spec but promoted as a column for filtering.
 	User string `gorm:"index" valid:"length(0|255)"`
+	// GORM doesn't save the zero value for ints, so we use a pointer for the State field
+	State *int32 `gorm:"default:0"`
+	// State updated at to indicate the archival/activation time
+	StateUpdatedAt *time.Time
```

Comment:
nit: does this need to be a model column? can we just store it in the execution closure?

Reply:
Thought was to support faster range based queries for state, but we can always add it later as a separate column when the need arises. I have currently kept this only in the closure

Reply:
I still see it in the model definition